app/jobs/create_virtual_objects_job.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+##
+# job to create virtual objects
+class CreateVirtualObjectsJob < GenericJob
+  queue_as :default
+
+  ##
+  # A job that creates virtual objects
+  # @param [Integer] bulk_action_id GlobalID for a BulkAction object
+  # @param [Hash] params additional parameters that an Argo job may need
+  # @option params [String] :create_virtual_objects CSV string
+  def perform(bulk_action_id, params)
+    super
+
+    virtual_objects = CsvToJsonService.convert(csv_string: params[:create_virtual_objects])
+
+    with_bulk_action_log do |log|
+      log.puts("#{Time.current} Starting #{self.class} for BulkAction #{bulk_action_id}")
+
+      parent_druids = virtual_objects[:virtual_objects].map { |hash| hash[:parent_id] }
+
+      # NOTE: We use this instead of `update_druid_count` because virtual object
+      #       creation does not use the `pids` form field.
+      bulk_action.update(druid_count_total: parent_druids.length)
+
+      # rubocop:disable Metrics/LineLength
+      problematic_druids = find_and_authorize_druids(parent_druids)
```

Comment:
Is it more normal for a method named `find_and_authorize_druids` to return the approved druids rather than the errors?  `problematic_druids` would be a better name, no?  and if you can memoize, you lose a line of code.

Reply:
I'll make these changes.